src/js/flows/refreshPoolNames.ts

```diff
@@ -6,10 +6,21 @@ import {getZealot} from "./getZealot"
 export default function refreshPoolNames(): Thunk<Promise<void>> {
   return (dispatch, getState) => {
     const zealot = dispatch(getZealot())
-    return zealot.pools.list().then((pools) => {
-      pools = pools || []
-      const id = Current.getWorkspaceId(getState())
-      if (id) dispatch(Pools.setPools(id, pools))
-    })
+    let pools = []
+    return zealot.pools
+      .list()
+      .then((data) => {
+        pools = data || []
+        return Promise.all(
+          pools.map(async (pool, i) => {
+            let stats = await zealot.pools.stats(pool.id)
+            pools[i] = {...pool, ...stats}
+          })
+        )
+      })
```

Comment:
You don't need to make the calls for all the stats here. The pool config object array will work fine when being passed into `Pools.setPools()`. It's a bad name, but it will merge the data with whats already there. And the app knows that calling `refreshPoolNames()` doesn't update all the pools stats.

Reply:
You do actually James. If this isn't called the pools in state don't have span info and when you click on the pool it shows the pool doesn't have any data. Probably something to revisit in another pr.